cg/meta/demultiplex/housekeeper_storage_functions.py

```diff
@@ -0,0 +1,219 @@
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.models.demultiplex.flow_cell import FlowCellDirectoryData
+from cg.constants.housekeeper_tags import SequencingFileTag
+from cg.constants.sequencing import Sequencers
+from cg.store import Store
+from cg.meta.demultiplex.utils import (
+    get_lane_from_sample_fastq,
+    get_q30_threshold,
+    get_sample_fastqs_from_flow_cell,
+    get_sample_sheet_path,
+)
+from cg.apps.demultiplex.sample_sheet.read_sample_sheet import (
+    get_sample_internal_ids_from_sample_sheet,
+)
+from cg.utils.files import get_files_matching_pattern
+
+LOG = logging.getLogger(__name__)
+
+
+def store_flow_cell_data_in_housekeeper(
+    flow_cell: FlowCellDirectoryData,
+    hk_api: HousekeeperAPI,
+    flow_cell_run_dir: Path,
+    store: Store,
+) -> None:
+    LOG.info(f"Add flow cell data to Housekeeper for {flow_cell.id}")
+
+    add_bundle_and_version_if_non_existent(bundle_name=flow_cell.id, hk_api=hk_api)
+
+    tags: List[str] = [SequencingFileTag.FASTQ, SequencingFileTag.SAMPLE_SHEET, flow_cell.id]
+    add_tags_if_non_existent(tag_names=tags, hk_api=hk_api)
+
+    add_sample_sheet_path_to_housekeeper(
+        flow_cell_directory=flow_cell.path, flow_cell_name=flow_cell.id, hk_api=hk_api
+    )
+    add_sample_fastq_files_to_housekeeper(flow_cell=flow_cell, hk_api=hk_api, store=store)
+    add_demux_logs_to_housekeeper(
+        flow_cell=flow_cell, hk_api=hk_api, flow_cell_run_dir=flow_cell_run_dir
+    )
+
+
+def add_demux_logs_to_housekeeper(
+    flow_cell: FlowCellDirectoryData, hk_api: HousekeeperAPI, flow_cell_run_dir: Path
+) -> None:
+    """Add demux logs to Housekeeper."""
+    log_file_name_pattern: str = r"*_demultiplex.std*"
+    demux_log_file_paths: List[Path] = get_files_matching_pattern(
+        directory=Path(flow_cell_run_dir, flow_cell.full_name), pattern=log_file_name_pattern
+    )
+
+    tag_names: List[str] = [SequencingFileTag.DEMUX_LOG, flow_cell.id]
+    for log_file_path in demux_log_file_paths:
+        try:
+            add_file_to_bundle_if_non_existent(
+                file_path=log_file_path,
+                bundle_name=flow_cell.id,
+                tag_names=tag_names,
+                hk_api=hk_api,
+            )
+            LOG.info(f"Added demux log file {log_file_path} to Housekeeper.")
+        except FileNotFoundError as e:
+            LOG.error(f"Cannot find demux log file {log_file_path}. Error: {e}.")
+
+
+def add_sample_fastq_files_to_housekeeper(
+    flow_cell: FlowCellDirectoryData, hk_api: HousekeeperAPI, store: Store
+) -> None:
+    """Add sample fastq files from flow cell to Housekeeper."""
+
+    sample_internal_ids: List[str] = get_sample_internal_ids_from_sample_sheet(
+        sample_sheet_path=flow_cell.sample_sheet_path,
+        flow_cell_sample_type=flow_cell.sample_type,
+    )
+
+    for sample_internal_id in sample_internal_ids:
+        add_bundle_and_version_if_non_existent(bundle_name=sample_internal_id, hk_api=hk_api)
```

Comment:
This needs to be removed as to not create empty sample bundles

Reply:
removed it now